src/Rings/mpoly-ideals.jl

```diff
@@ -1733,10 +1747,23 @@ false
 ```
 """
 @attr Bool function is_prime(I::MPolyIdeal)
+  is_known(I, primary_decomposition) && is_one(length(primary_decomposition(I))) && return is_subset(first(primary_decomposition(I))[2], I)
   D = minimal_primes(I)
   return length(D) == 1 && issubset(D[1], I)
 end
 
+function is_known(I::MPolyIdeal, ::typeof(is_prime))
+  has_attribute(I, :is_prime) && return true
+  is_known(I, primary_decomposition) && return true
+  is_known(I, minimal_primes) && return true
+  if coefficient_ring(base_ring(I)) isa Field
```

Comment:
Wouldn't it suffice if the base_ring is a domain, as echeck via `is_domain_type` ? I.e.
```suggestion
  if is_domain_type(base_ring_type(I))
```

Reply:
You are probably right. This might be a reasonable extension. Just didn't occur to me when I wrote it up.